portal-web/test/functional/com/liferay/portalweb/macros/LiferayEvent.macro

```diff
@@ -75,4 +75,31 @@ window.LFR_JS_EVENTS_LOG = {
 		RunScript(locator1 = "${javascript}");
 	}
 
+	macro waitForLiferayEvent {
+		var javascript = '''
+try {
+	var eventValue = LFR_JS_EVENTS_LOG.getLastEvent("${eventName}").${attributeName};
+
+	if (eventValue.includes("${attributeValue}")) {
+		return "fired";
+	}
+	else {
+		return "pending";
+	}
+}
+catch(err) {
+	return "pending";
+}
+		''';
+		var liferayEventFired = "";
+
+		while ("${liferayEventFired}" != "fired" && (maxIterations = "5")) {
+			var liferayEventFired = selenium.getEval("${javascript}");
+
+			echo("${liferayEventFired}");
+
+			Pause(locator1 = "1000");
```

Comment:
If I understand this correctly, we're pausing for 1000ms for a maximum number of 5 times.

Assuming a uniform distribution, this means we're needlessly waiting 500ms on average every time we call this.

Is there a reason why the pause is so big? How costly it is for the thread or the jvm to sleep and wake up that grants a whole second? Is there any data to back up these numbers?

Why don't we pause for 100ms a maximum of 50 times? That way, on average, we're only wasting 50ms and tests will run pass this point half a second faster on average.

One thing I observed watching POSHI run locally is that there's a huge amount of idle time. It was really noticeable when filling out inputs.Execution would be idle for seconds after the input was already filled up before moving onto the next step. Is it possible that these pauses have been slowly crawling and growing over time to account for "test flakiness"? Seems like an improvement area worth analyzing...

Reply:
On top of what I said above... won't this make a successful condition wait for an additional 1000 seconds? shouldn't this Pause be wrapped in a conditional to check for the exit condition?

Reply:
This is all up for change, but I was trying to match the Condition logic in the Java side, which will poll, wait 1 sec until it times out after 2 minutes. In this specific case, I changed it to 5 times to make it fail faster locally when testing and forgot to change it back lol. 

It could probably be improved and that possibly means it's slower locally, but portal on CI is pretty slow in my observations. Especially on first start ups, but maybe @john-co or @vicnate5 can weigh in more on that. 

We can change the overall wait polling logic timing if that's a big priority though. There's a lot of improvements to be made, but I don't think that's something to explore right now.